server/py/framework/utils/singletons/k8s.py

```diff
@@ -580,22 +582,45 @@ def store_secrets(
         return mlrun.common.schemas.SecretEventActions.updated
 
     def read_secret(
-        self, secret_name: str, namespace: str = "", silent=False
+        self,
+        secret_name: str,
+        namespace: str = "",
+        labels: typing.Optional[dict[str, str]] = None,
+        silent=False,
     ) -> typing.Optional[client.V1Secret]:
         namespace = self.resolve_namespace(namespace)
         if not silent:
-            logger.debug("Reading secret", secret_name=secret_name, namespace=namespace)
+            logger.debug(
+                "Reading secret",
+                secret_name=secret_name,
+                namespace=namespace,
+                labels=labels,
+            )
         try:
             k8s_secret = self.v1api.read_namespaced_secret(
                 name=secret_name,
                 namespace=namespace,
             )
+            if labels:
+                secret_labels = k8s_secret.metadata.labels or {}
+                for key, value in labels.items():
+                    if secret_labels.get(key) != value:
+                        if not silent:
+                            logger.debug(
+                                "Secret found but labels did not match",
+                                secret_name=secret_name,
+                                expected_labels=labels,
+                                actual_labels=secret_labels,
+                            )
+                        return None
```

Comment:
```suggestion
    # Find any label that does not match
        mismatched = {k: (v, secret_labels.get(k)) for k, v in labels.items() if secret_labels.get(k) != v}
        if mismatched:
            if not silent:
                logger.debug(
                    "Secret found but labels did not match",
                    secret_name=secret_name,
                    expected_labels=labels,
                    actual_labels=secret_labels,
                    mismatched_labels=mismatched,
                )
            return None
```
to show all mismatched in a one shot